extension/httpfs/include/httpfs.h

```diff
@@ -0,0 +1,92 @@
+#pragma once
+
+#include "common/file_system/file_system.h"
+#include "httplib.h"
+
+namespace kuzu {
+namespace httpfs {
+
+using HeaderMap = std::unordered_map<std::string, std::string>;
+
+struct HTTPResponse {
+public:
+    HTTPResponse(httplib::Response& res, const std::string& url);
+    int code;
+    std::string error;
+    HeaderMap headers;
+    std::string url;
+    std::string body;
+};
+
+struct HTTPParams {
+    // TODO(Ziyi): Make them configurable.
+    static constexpr uint64_t DEFAULT_TIMEOUT = 30000; // 30 sec
+    static constexpr uint64_t DEFAULT_RETRIES = 3;
+    static constexpr uint64_t DEFAULT_RETRY_WAIT_MS = 100;
+    static constexpr float DEFAULT_RETRY_BACKOFF = 4;
+    static constexpr bool DEFAULT_KEEP_ALIVE = true;
+};
+
+struct HTTPFileInfo : public common::FileInfo {
+    HTTPFileInfo(std::string path, common::FileSystem* fileSystem, int flags);
+
+    // We keep a http client stored for connection reuse with keep-alive headers
+    std::unique_ptr<httplib::Client> httpClient;
```

Comment:
You probably don't need to use unique_ptr here

Reply:
no, it doesn't have a default constructor, and we are not able to init it during initialization